### PR TITLE
Glossary

### LINKED ISSUE #4572: Glossary
People that discover OC can be struggling at first at understanding some concepts.

Here is a list of terms that you can find in OC and a small explanation.
These are just the ones that a Power user or a Theme designer could see, so there won't be any concepts for Application and Modules developers.
This could be the start for some kind of Glossary that could be added in docs.

Administrator:
**Deployment Step**: Export of a specific set of information (Configuration or Data)
**Deployment Plan**: Batch Export that executes multiple deployment steps
**Recipe**: Json file used to import Wizard with different steps to execute
**Setup Recipe**: Import steps like Set theme, Define types, Import data, ... executed during setup 
**Permission**: Allow a Users in a Role to do a have access to a specific action

Power user:
**Content Part**: Extends a content type with a complex piece. There can be only one part of the same kind by content type.
**Display type**: A way to specify in which context a content is rendered (ex: Details, Summary, SummaryAdmin). A content can be displayed differently depending on which Display type is rendered.
**Content Field**: Extend a content type with a simple piece. There can be multiple fields of the same kind attached to a content type.
**Field Editor**: A field can have different Editors (ex: The value of a Numeric Field can be set by an input or a slider)
**Autoroute**: Custom Permalink used for Routing. It can use a Liquid pattern to be automatically generated.
**Bag**: Collection of content items of a certain type in a parent content item
**List**: To attach content items to a parent Container (ex: A blog contains a list of blog post)
**Taxonomy**: Category that contains specific terms
**Admin menu**: Custom link added in the left admin menu
**Alias**: A way to identify an item with a key that you can call to retrieve it, instead of an Id
**Content Preview**: Allows to Preview and Live Edit a content
**Indexing**: Define the way the content will be indexed in order to be search it from a query
**Query**: Parameterized Lucene or Sql query defined in admin
**Tenant**: Independant subsite. One OC instance can have multiple tenants. They can only be managed in the Default one.

Theme Designer:
**Theme**: Template with assets and views used to customize the display
**Liquid**: Templating Code Syntax that you can use instead of Razor.
**Alternate**: Override of content type or part or field using a file in a Theme
**Placement**: Mapping file to set the order of appearance or hide contents for a specific content type or Part/Field name or display type in a Theme.
**Assets**: In the Admin, this is the Media library. In a Theme, these are the folders in wwwroot.
**Resource**: Style or Script libraries that are registered with a specific version and with potentially minified file and CDN urls
**Shape**: https://github.com/OrchardCMS/OrchardCore/issues/4121#issuecomment-539608731

**Template**: Liquid Override of a Shape or a Display type in admin
**Zone**: Section in the Layout (ex: Footer) in which you can render items
**Layer**: Display Rule in which you specify a condition to be rendered (ex: isHomepage())
**Widget**: Display Content in a specific zone and layer
**Flow**: Page Layout in which you can add widgets

If you want to add/edit some terms and descriptions or ask for explanations for some concepts, please add a comment.

### PR DESCRIPTION
**Pull request by agriffard**
Fixes #4572

### CODE REVIEW

**Comment by agriffard** [comment]
Added some concepts appearing in http://docs.orchardproject.net/en/latest/Documentation/Basic-Orchard-Concepts/

**Comment by agriffard** [comment]
You can see a preview here: https://orchardcore.readthedocs.io/en/ag-glossary/docs/glossary/

**Comment by agriffard** [comment]
How would you explain this concept?
Stereotype

**Comment by deanmarcussen** [comment]
> How would you explain this concept?
> Stereotype

Hard to describe. I would do it via examples.
i.e.
By default Content Items have no stereotype, however certain modules will use a defined StereoType to determine which content items can be used by them. Examples of this include the Menu Stereotype, and the Widget Stereotype.

**Comment by Unconquered1** [comment]
### Content Item - Proposed Definition

The final version of a piece of web content that can be published for display via a web browser. It is created in the OrchardCore Admin GUI from a content type. It can be versioned, localized, and can have its own unique URL (depending on its Content Parts). The display of a Content Item can be restricted by a user’s assigned permissions via a role or otherwise. A few examples are a web page, a blog post, or a product listing.

**Comment by Unconquered1** [comment]
### Autoroute - Proposed Definition
A Content Part that can be added to a Content Type to provide a dynamic URL feature.  This allows any Content Item built from the resulting Content Type to be directly accessible via a web address. An Autoroute can use a Liquid pattern to automatically generate the desired URL.

**Comment by Unconquered1** [comment]
### Shape - Proposed Definition (WIP)
An object in OrchardCore used to render HTML content. A Shape defines the type of content to be rendered, it receives its instructions on how to display the content it will render from a template. This makes the shape able to dynamically adapt the way it displays its content, based on a chosen OrchardCore theme.

File: src/docs/glossary/README.md [hunk 1]
```diff
@@ -136,6 +136,11 @@ A section in the Layout (ex: Footer) in which you can render items.
 A display Rule in which you specify a condition to be rendered (ex: isHomepage()).  
 See [Layers](../reference/modules/Layers)
 
+### Stereotype
+
+By default Content Items have no stereotype, however certain modules will use a defined StereoType to determine which content items can be used by them.  
+Examples of this include the Menu Stereotype, and the Widget Stereotype.
+
 ### Widget
```
Comment (Skrypt) [inline src/docs/glossary/README.md]: which content type items.

Comment (agriffard) [reply src/docs/glossary/README.md]: `which content types`

Comment (Skrypt) [reply src/docs/glossary/README.md]: works too

**Pull request merged**